cpp/src/arrow/acero/hash_join_node_test.cc

```diff
@@ -3370,8 +3370,10 @@ TEST(HashJoin, LARGE_MEMORY_TEST(BuildSideOver4GBVarLength)) {
   constexpr int value_no_match_length_min = 128;
   constexpr int value_no_match_length_max = 129;
   constexpr int value_match_length = 130;
+  // The value "DDD..." will be hashed to the partition over 4GB of the hash table.
```

Comment:
Note this really depends on the hash algorithm which is opaque, but this as well is the best I can do so far.